### PR TITLE
adding azure firewall resource

### PR DESCRIPTION
**Pull request by Rohit1509**
### Description

Adding Azure Firewall Singular and Plural resource.

### Issues Resolved

N/A

### Check List

- [x] New functionality includes tests
- [ ] All tests pass
- [ ] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/azure_firewall.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  def initialize(opts = {})
+    Inspec::Log.warn Helpers.resource_deprecation_message(@__resource_name__, AzureFirewall.name)
+    # Options should be Hash type. Otherwise Ruby will raise an error when we try to access the keys.
+    raise ArgumentError, 'Parameters must be provided in an Hash object.' unless opts.is_a?(Hash)
+
+    # For backward compatibility.
+    opts[:api_version] ||= '2019-05-01'
+    super
+  end
+end
```
Comment (sathish-progress) [inline libraries/azure_firewall.rb:48]: this is not needed for new resources!

File: libraries/azure_firewalls.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
+require 'azure_generic_resources'
+
+class AzureContainerRegistries < AzureGenericResources
```
Comment (sathish-progress) [inline libraries/azure_firewalls.rb:3]: this whole file is wrong!!

**Comment by soumyo13** [comment]
@Rohit1509 Is this PR still needed or we can close it?

**Comment by Rohit1509** [comment]
Hi @soumyo13, apologies for late reply. This PR is no longer needed and can be closed out. Thanks!

**Comment by sathish-progress** [comment]
closing this based on above comment

**Pull request closed without merging**